lib/editor-linter.coffee

```diff
@@ -53,30 +53,34 @@ class EditorLinter
 
     scopes = @editor.scopeDescriptorForBufferPosition(@editor.getCursorBufferPosition()).scopes
     scopes.push '*' # To allow global linters
-
-    Promise.all(@triggerLinters(wasTriggeredOnChange, scopes)).then =>
+    @triggerLinters(true, wasTriggeredOnChange, scopes).then( =>
+      return Promise.all(@triggerLinters(false, wasTriggeredOnChange, scopes))
+    ).then =>
       @lock(wasTriggeredOnChange, false)
 
   # This method returns an array of promises to be used in lint
-  triggerLinters: (wasTriggeredOnChange, scopes) ->
-    Promises = []
+  triggerLinters: (bufferModifying, wasTriggeredOnChange, scopes) ->
+    ToReturn = if bufferModifying then Promise.resolve() else []
```

Comment:
It's not a good idea to have a variable that alternates between different types. Sometimes it's a `Promise`, sometimes it's an `Array`. Off the top of my head I don't know what the exact solution is here, but I don't think you should combine these.